metrics_proto.go

```diff
@@ -635,10 +624,25 @@ func labelDescriptorsFromProto(defaults map[string]labelValue, protoLabelKeys []
 	return labelDescriptors
 }
 
-func (se *statsExporter) metricTypeFromProto(name string) (string, bool) {
-	// TODO: (@odeke-em) support non-"custom.googleapis.com" metrics names.
-	name = path.Join("custom.googleapis.com", "opencensus", name)
-	return name, true
+func (se *statsExporter) metricTypeFromProto(name string) string {
+	prefix := se.o.MetricPrefix
+	if prefix != "" {
+		name = prefix + name
+	}
+	if !hasDomain(name) {
+		name = path.Join("custom.googleapis.com", "opencensus", name)
```

Comment:
Can we then pre-calculate:
se.defaultDomain = path.Join("custom.googleapis.com", "opencensus")

if !hasDomain(name) {
	// Still needed because the name may or may not have a "/" at the beginning.
	name = path.Join(se.defaultDomain, name)
}

Reply:
SG, updated.